# Velocities
float32[] linear_vel
float32[] angular_vel

# Contact states
bool[] toes_contact
bool[] thighs_contact
bool[] shanks_contact

# Joints states
float32[] joint_angles
float32[] joint_velocities

# Transformation matrices
float32[] transf_matrix0
float32[] transf_matrix1
float32[] transf_matrix2
float32[] transf_matrix3